phpBB/phpbb/template/twig/extension/icon.php

```diff
@@ -0,0 +1,255 @@
+	 * @param array			$attributes		Additional attributes for the icon, where the key is the attribute.
+	 *                      				{'data-ajax': 'mark_forums'} results in ' data-ajax="mark_forums"'
+	 * @return string
+	 */
+	public function icon(environment $environment, $type, $icon, $title = '', $hidden = false, $classes = '', array $attributes = [])
+	{
+		$type = strtolower($type);
+
+		if (empty($icon))
+		{
+			return '';
+		}
+
+		$not_found = false;
+		$source = '';
+
+		switch ($type)
+		{
+			case 'font':
+				// Nothing to do here..
+			break;
+
+			case 'iconify':
+				$source = explode(':', $icon);
+				$source = $source[0];
+			break;
+
+			case 'png':
+				$filesystem	= $environment->get_filesystem();
+				$root_path	= $environment->get_web_root_path();
+
+				$board_url	= defined('PHPBB_USE_BOARD_URL_PATH') && PHPBB_USE_BOARD_URL_PATH;
+				$base_path	= $board_url ? generate_board_url() . '/' : $root_path;
+
+				// Iterate over the user's styles and check for icon existance
+				foreach ($this->get_style_list() as $style_path)
+				{
+					if ($filesystem->exists("{$root_path}styles/{$style_path}/theme/icons/png/{$icon}.png"))
+					{
+						$source = "{$base_path}styles/{$style_path}/theme/icons/png/{$icon}.png";
+
+						break;
+					}
+				}
+
+				// Check if the icon was found or not
+				$not_found = empty($source);
+			break;
+
+			case 'svg':
+				try
+				{
+					// Try to load and prepare the SVG icon
+					$file	= $environment->load('icons/svg/' . $icon . '.svg');
+					$source	= $this->prepare_svg($file);
+				}
+				catch (\Twig\Error\LoaderError $e)
+				{
+					// Icon was not found
+					$not_found = true;
+				}
+				catch (\Twig\Error\Error $e)
+				{
+					return '';
+				}
+			break;
+
+			default:
+				return '';
+			break;
+		}
+
+		// If no PNG or SVG icon was found, display a default 404 PNG icon.
+		if ($not_found)
+		{
+			if (empty($base_path))
+			{
+				$board_url = defined('PHPBB_USE_BOARD_URL_PATH') && PHPBB_USE_BOARD_URL_PATH;
+				$base_path = $board_url ? generate_board_url() . '/' : $environment->get_web_root_path();
+			}
+
+			$source = "{$base_path}styles/all/imgs/icons/png/404.png";
+			$type = 'png';
+			$icon = '404';
+		}
+
+		try
+		{
+			return $environment->render("macros/icons/{$type}.html", [
+				'ATTRIBUTES'	=> (string) $this->implode_attributes($attributes),
+				'CLASSES'		=> (string) $classes,
+				'ICON'			=> (string) $icon,
+				'SOURCE'		=> (string) $source,
+				'TITLE'			=> (string) $title,
+				'S_HIDDEN'		=> (bool) $hidden,
+			]);
+		}
+		catch (\Twig\Error\Error $e)
+		{
+			return '';
+		}
+	}
+
+	/**
+	 * Prepare an SVG for usage in the template icon.
+	 *
+	 * This removes any <?xml ?> and <!DOCTYPE> elements,
+	 * aswell as any <svg> and <title> elements.
+	 *
+	 * @param \Twig\TemplateWrapper	$file	The SVG file loaded from the environment
+	 * @return string						The cleaned SVG
+	 */
+	protected function prepare_svg(\Twig\TemplateWrapper $file)
```

Comment:
This is messing up the viewBox parameter. Which needs to be pulled from the original source. Somehow you have it defaulting to "0 0 24 24" which is incorrect